collector/iscsi_linux.go

```diff
@@ -0,0 +1,402 @@
+			[]string{"iqn", "tpgt", "lun", "iblock", "object", "blockname"}, nil,
+		),
+		lioBlockWrite: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioIblockSubsystem, "write_total"),
+			"iSCSI IBlock backstore Write in byte.",
+			[]string{"iqn", "tpgt", "lun", "iblock", "object", "blockname"}, nil,
+		),
+
+		lioRbdIops: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRbdSubsystem, "iops_total"),
+			"iSCSI RBD backstore transport operations.",
+			[]string{"iqn", "tpgt", "lun", "rbd", "pool", "image"}, nil,
+		),
+		lioRbdRead: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRbdSubsystem, "read_total"),
+			"iSCSI RBD backstore Read in byte.",
+			[]string{"iqn", "tpgt", "lun", "rbd", "pool", "image"}, nil,
+		),
+		lioRbdWrite: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRbdSubsystem, "write_total"),
+			"iSCSI RBD backstore Write in byte.",
+			[]string{"iqn", "tpgt", "lun", "rbd", "pool", "image"}, nil,
+		),
+
+		lioRdmcpIops: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "iops_total"),
+			"iSCSI Memory Copy RAMDisk backstore transport operations.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+		lioRdmcpRead: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "read_total"),
+			"iSCSI Memory Copy RAMDisk backstore Read in byte.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+		lioRdmcpWrite: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "write_total"),
+			"iSCSI Memory Copy RAMDisk backstore Write in byte.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+	}, nil
+}
+
+// /sys/kernel/config/target/iscsi/iqn*/tpgt_*/lun/lun_*/ which link
+// back to the following
+// /sys/kernel/config/target/core/{backstoreType}_{number}/{objectName}/
+func (c *lioCollector) updateStat(ch chan<- prometheus.Metric, s *iscsi.Stats) error {
+
+	log.Debugf("lio updateStat iscsi %s path", s.Name)
+	tpgtS := s.Tpgt
+	for _, tpgt := range tpgtS {
+		tpgtPath := tpgt.TpgtPath
+		iscsiEnable := tpgt.IsEnable
+
+		log.Debugf("lio: iscsi %s isEnable=%t", tpgtPath, iscsiEnable)
+		// let's not putting more line into the graph with multiple
+		// disable lun, it may create problem for bigger cluster
+		if iscsiEnable {
+
+			lunS := tpgt.Luns
+			for _, lun := range lunS {
+				backstoreType := lun.Backstore
+				objectName := lun.ObjectName
+				typeNumber := lun.TypeNumber
+
+				// struct type graphLabel { iqn, tpgt, lun, store, pool,  image}
+				// label := graphLabel {iqn, tpgt, lun, backstoreType, objectName, typeNumber}
+				label := graphLabel{s.Name, tpgt.Name, lun.Name, backstoreType, objectName, typeNumber}
+
+				log.Debugf("lio: iqn=%s, tpgt=%s, lun=%s, type=%s, object=%s, typeNumber=%s",
+					s.Name, tpgt.Name, lun.Name, backstoreType, objectName, typeNumber)
+
+				switch {
+				case strings.Compare(backstoreType, "fileio") == 0:
```

Comment:
I think `strings.Compare` here is not necessary, you can just use `case backstoreType == "fileio"`.

Reply:
Yeah, I guess I'm not fully knowing the different of using "string" == compare vs strings.Compare here so I pick the function call. If they are the same, I will switch all to ==. Thanks